### PR TITLE
dotnet 6.0.109

### PR DESCRIPTION
**Pull request by cho-m**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/dotnet.rb [hunk 1]
```diff
@@ -92,11 +90,6 @@ def install
 
     (buildpath/"src/SourceBuild/tarball/patches/msbuild").install resource("homebrew-msbuild-patch")
 
-    # Fix usage of GNU-specific flag.
-    # TODO: Remove this when upstreamed
-    inreplace "src/SourceBuild/tarball/content/repos/Directory.Build.targets",
-              "--block-size=1M", "-m"
-
     Dir.mktmpdir do |sourcedir|
       system "./build.sh", "/p:ArcadeBuildTarball=true", "/p:TarballDir=#{sourcedir}"
```
Comment (cho-m) [inline Formula/dotnet.rb:94]: From local attempt, ARM build is failing as it is trying to get x64 runtime.

Comment (cho-m) [reply Formula/dotnet.rb:94]: ARM build breakage is probably due to https://github.com/dotnet/install-scripts/pull/314. Manually downloading previous copy of script no longer tries to download x64 tarballs.

**Comment by cho-m** [comment]
Linux failure is probably https://github.com/dotnet/runtime/commit/992cf8c97cc71d4ca9a0a11e6604a6716ed4cefc. Will add patch and rerun.

File: Formula/dotnet.rb [hunk 2]
```diff
@@ -72,63 +64,85 @@ class Dotnet < Formula
...
+      # Issue ref: https://github.com/dotnet/source-build/issues/2795
+      inreplace "src/installer/src/redist/targets/GenerateLayout.targets",
                 "$(MicrosoftAspNetCoreAppRuntimePackageVersion)",
                 "$(MicrosoftAspNetCoreAppRuntimewinx64PackageVersion)"
 
       # Rename patch fails on case-insensitive systems like macOS
       # TODO: Remove whenever patch is no longer used
-      rm Dir["src/nuget-client.*/eng/source-build-patches/0001-Rename-NuGet.Config*.patch"].first if OS.mac?
+      rename_patch = "0001-Rename-NuGet.Config-to-NuGet.config-to-account-for-a.patch"
+      (Pathname.pwd/"src/nuget-client/eng/source-build-patches"/rename_patch).unlink if OS.mac?
```
Comment (carlocab) [inline Formula/dotnet.rb:134]:
```suggestion
      (Pathname("src/nuget-client/eng/source-build-patches")/rename_patch).unlink if OS.mac?
```
For next time, maybe.

**Review by carlocab — APPROVED** [review]
Nice.

**Pull request closed without merging**